src/transform-microsyntax.ts

```diff
@@ -186,28 +187,54 @@ export function transformTemplateBindings(
     } as T & RawNGSpan;
   }
 
-  function findBackKeySpan(start: number, key: string): RawNGSpan {
-    if (context.text[start] !== "'" && context.text[start] !== '"') {
-      return { start, end: start + key.length };
-    }
-    const quote = context.text[start];
-    let backslash = 0;
-    let index = start + 1;
-    while (true) {
-      const char = context.text[index];
-      if (char === quote && backslash % 2 === 0) {
-        return { start, end: index + 1 };
-      }
-      if (char === '\\') {
-        backslash++;
-      } else {
-        backslash = 0;
-      }
-      index++;
+  function removePrefix(string: string) {
+    return toLowerCamelCase(string.slice(prefix.source.length));
+  }
+
+  function isExpressionBinding(
+    templateBinding: ng.TemplateBinding,
+  ): templateBinding is ng.ExpressionBinding {
+    return templateBinding instanceof ng.ExpressionBinding;
+  }
+
+  function isVariableBinding(
+    templateBinding: ng.TemplateBinding,
+  ): templateBinding is ng.VariableBinding {
+    return templateBinding instanceof ng.VariableBinding;
+  }
+
+  function fixTemplateBindingSpan(templateBinding: ng.TemplateBinding) {
+    fixSpan(templateBinding.key.span);
+    if (isVariableBinding(templateBinding) && templateBinding.value) {
+      fixSpan(templateBinding.value.span);
     }
   }
 
-  function removePrefix(string: string) {
-    return toLowerCamelCase(string.slice(prefix.length));
+  /**
+   * - "a"  (start=0 end=1) -> (start=0 end=3)
+   * - '\'' (start=0 end=1) -> (start=0 end=4)
+   */
+  function fixSpan(span: RawNGSpan) {
+    if (context.text[span.start] !== '"' && context.text[span.start] !== "'") {
+      return;
+    }
+    const quote = context.text[span.start];
+    let hasBackSlash = false;
+    for (let i = span.start + 1; i < context.text.length; i++) {
+      switch (context.text[i]) {
+        case quote:
+          if (!hasBackSlash) {
+            span.end = i + 1;
+            return;
+          }
+        // fall through
+        default:
+          hasBackSlash = false;
+          break;
+        case '\\':
+          hasBackSlash = !hasBackSlash;
+          break;
```

Comment:
Is this branch  dead code ?

Reply:
Can I put a default between cases?
Yes, you can! JavaScript will drop you back to the default if it can't find a match:

never thought  about this  before...